### PR TITLE
fixed compiler errors

### PR DESCRIPTION
**Pull request by Aman25ta**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
+    "@typescript-eslint/parser": "^4.28.2",
+    "eslint": "^7.30.0",
+    "eslint-config-airbnb-typescript": "^12.3.1"
   },
-  "author": "",
-  "license": "ISC",
-  "description": ""
+  "license": "GPLv3",
+  "scripts": {
+    "start": "tsc && echo 'Built.' && node build/index.js"
```
Comment (AltF02) [inline package.json]: You can remove the echo here, tsc already shows if it's successful or not

File: src/commands/nickreq/nickreq.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import { Command, CommandoClient, CommandoMessage } from 'discord.js-commando';
+import { load } from 'js-yaml';
+import { readFileSync } from 'fs';
+import { MessageEmbed, TextChannel } from 'discord.js';
+import DB from '../../db.js';
+
+const file = readFileSync('./config.yml', 'utf8');
+const config: any = load(file)!;
```
Comment (AltF02) [inline src/commands/nickreq/nickreq.ts]: Please don't use any, I recommend making a config interface, take a look at the other projects like modmail for examples

File: src/commands/nickreq/nickreq.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        {
+          key: 'nick',
+          prompt: 'No nickname provided, please enter a nickname.',
+          type: 'string',
+        },
+      ],
+    });
+  }
+
+  async run(message: CommandoMessage, { nick }:{ nick: string }): Promise<null> {
```
Comment (AltF02) [inline src/commands/nickreq/nickreq.ts]: Nit: space between `{ nick }: { nick:string }`

File: src/commands/nickreq/nickreq.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    if (message.guild == null) return null;
+    if (nick.length > 32) {
+      await message.reply('The nickname must be less than 32 characters');
+      return null;
+    }
+
+    const re = /^[\\x00-\\x7F]/;
+    const testx = re.test(nick);
+    if (testx === true) {
```
Comment (AltF02) [inline src/commands/nickreq/nickreq.ts]: This can just be

```ts
if (testx) {
		//...
}
```

File: src/db.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import { Pool } from 'pg';
+import { load } from 'js-yaml';
+import { readFileSync } from 'fs';
+
+const file = readFileSync('./config.yml', 'utf8');
+const config: any = load(file)!;
```
Comment (AltF02) [inline src/db.ts]: Same as before

File: src/index.ts [hunk 6]
```diff
@@ -1,27 +1,96 @@
...
-
-dotenv.config();
-
-const client = new CommandoClient({
-  commandPrefix: '.nick',
-  presence: {
-    status: "dnd"
-  
-  }
+const commando = require('discord.js-commando');
```
Comment (AltF02) [inline src/index.ts]: Please just do
`import * as commando from 'discord.js-commando'`

File: src/index.ts [hunk 7]
```diff
@@ -1,27 +1,96 @@
...
+import { join } from 'path';
+import { load } from 'js-yaml';
+import { readFileSync } from 'fs';
+import DB from './db.js';
```
Comment (AltF02) [inline src/index.ts]: Remove the extension here

**Review by AltF02 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (AltF02) [reply package.json]: On top of this, you can just do `node build` as it will automatically get the index.js

File: src/index.ts (hunk 6 above)
Comment (Aman25ta) [reply src/index.ts]: I left this as require on purpose, as the library had problems with import here too

Comment (AltF02) [reply src/index.ts]: I think its still possible, give it a try and tell me how it goes

Comment (AltF02) [reply src/index.ts]: https://github.com/NewCircuit/nickreq/commit/ee374c4c3fffa75fd7df9357c5d7a74ff4246ef9#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R4

Comment (Aman25ta) [reply src/index.ts]: yep, same issue, import doesnt work with it

File: package.json [hunk 8]
```diff
@@ -9,11 +9,17 @@
...
     "eslint": "^7.30.0",
     "eslint-config-airbnb-typescript": "^12.3.1"
   },
-  "license": "GPLv3"
-}
+  "license": "GPL-3.0",
+  "scripts": {
+    "start": "tsc && node build"
+  }
+}
```
Comment (AltF02) [inline package.json]: NIt: No blank line at EOF

File: src/config.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+      if (!existsSync(Config.LOCATION)) {
+        throw new Error('Please create a config.yml');
+      }
+
+      const fileContents = readFileSync(
+        Config.LOCATION,
+        'utf-8',
+      );
+      const casted = load(fileContents) as Config;
+      return casted;
```
Comment (AltF02) [inline src/config.ts]: The variable `casted` is redundant, it can be 

```ts
return load(fileContents) as Config;
```

File: src/config.ts [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+export default class Config {
+    public readonly token: string;
+    public readonly channelid: string;
+    public readonly color: number;
+    public readonly ownerid: string;
+    public readonly pgusername: string;
+    public readonly pgpassword: string;
+    public readonly pghost: string;
+    public readonly pgport: number;
+    public readonly pgdbname: string;
```
Comment (AltF02) [inline src/config.ts]: Please use an interface for database, more info [here](https://www.typescriptlang.org/docs/handbook/interfaces.html)

**Review by AltF02 — CHANGES_REQUESTED** [review]
Final note, when trying to run eslint it doesn't work without typescript and devDependency, please add it with

`yarn add -D typescript`

**Comment by AltF02** [comment]
Sorry, there is a new feature requested:

> Banned words auto reject

This can be possibly be done by just having an txt file and a list of them and parsing that into an array. But you can also use the database because you've already set that up anyways. Add a command to list them, delete one, or add one

**Comment by Aman25ta** [comment]
> Sorry, there is a new feature requested:
> 
> 
> 
> > Banned words auto reject
> 
> 
> 
> This can be possibly be done by just having an txt file and a list of them and parsing that into an array. But you can also use the database because you've already set that up anyways. Add a command to list them, delete one, or add one

Sorry for the late reply, didn't notice this until now

Ill just use a txt file i guess because it might not make too much sense to have a db just for an array of banned words

File: src/commands/nickreq/remove.ts [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+          type: 'string',
+        },
+      ],
+    });
+  }
+
+  async run(message: CommandoMessage, { words }: { words: string }): Promise<null> {
+    const wordArray = words.split('\n').filter((e) => e !== '');
+    const bannedWords = readFileSync('./bannedwords.txt', 'utf-8').split('\n').filter((e) => e !== '');
+    const finalStr = bannedWords.filter((x) => !wordArray.includes(x)).join('\n');
```
Comment (AltF02) [inline src/commands/nickreq/remove.ts:27]: There's a dupe of code here and in add, I recommend adding a function to fetch the banned words

**Review by AltF02 — CHANGES_REQUESTED** [review]
Almost there :smile:

File: src/commands/nickreq/remove.ts (hunk 11 above)
Comment (Aman25ta) [reply src/commands/nickreq/remove.ts:27]: its just one line in both files though, is a seperate function really necessary?

Comment (AltF02) [reply src/commands/nickreq/remove.ts:27]: Its fine for now

**Pull request merged**